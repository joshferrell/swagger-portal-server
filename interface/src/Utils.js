export const promiseLogger = (x) => {
   console.log( x );
   return x;
}

export const match = (regex, text) =>
  new RegExp( regex ).test( text )

export const toJson = (fetchResponse) =>
  fetchResponse.json()
