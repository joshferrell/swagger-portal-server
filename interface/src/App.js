import React, { Component } from 'react';
import { HashRouter as Router, Route } from 'react-router-dom';
import './App.css';
import logo from './logo.svg';
import Search from './Search';
import Document from './Document';
import { toJson, promiseLogger, find, propEq } from './Utils';

export default class App extends Component {
    constructor(props) {
        super(props);
        this.state = { apis: [] };
    }

  renderDocument(request) {
      const { id } = request.match.params
      const doc = find( propEq( 'id', id ), this.state.apis )
      if ( doc ) {
        return <Document id={ id } title={ doc.title } />
      }
      else {
        return <h1>Document does not exist <a href="/">Try Another</a></h1>
      }
  }

  componentDidMount() {
    fetch( '/docs' )
      .then( toJson )
      .then( apis => this.setState( { apis } ) )
      .then( promiseLogger )
      .catch( error => this.setState( { message: "something went wrong" } ) )
  }

  render() {
    return (
      <Router>
        <div className="App">
          <div className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <h2>Welcome to React</h2>
          </div>
          <Route exact path="/" render={ () => <Search apis={ this.state.apis } /> } />
          <Route exact path="/:id" render={ this.renderDocument.bind(this) } />
        </div>
      </Router>
    );
  }
}
