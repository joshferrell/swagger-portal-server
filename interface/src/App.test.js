import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import { checkResults } from './Utils';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

it('should render the search view by default', () => {
    const component = renderer.create(<App />);
    expect(component.toJSON()).toMatchSnapshot();
});

it('should render the search view by default', () => {
    const sample = [{ title: 'foo' }, { title: 'bar' }];
    expect(checkResults('oo', sample)).toEqual([{ title: 'foo' }]);
});
