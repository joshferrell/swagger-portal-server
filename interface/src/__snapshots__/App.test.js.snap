exports[`test should render the search view by default 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg" />
    <h2>
      Welcome to React
    </h2>
  </div>
  <section>
    <form>
      <label>
        Filter API
      </label>
      <input
        onInput={[Function]}
        value="" />
    </form>
    <ul
      className="List" />
  </section>
</div>
`;
